### PR TITLE
Add new lmdb API csv_match_pop

### PR DESCRIPTION
**Pull request by rranjan3**
- Add new API to match value and pop from csv value
 - Update corresponding python interface and remote_lmdb as well
 - Add unit tests to verify csv_match_pop

Signed-off-by: Rajeev Ranjan <rajeev2.ranjan@intel.com>

### CODE REVIEW

File: shared_kv_storage/kv_storage/interface/kv_csv_interface.py [hunk 1]
```diff
@@ -70,3 +70,25 @@ def csv_pop(self, table, key):
...
         """
         pass
+
+# ---------------------------------------------------------------------------------------------------
+    @abstractmethod
+    def csv_match_pop(self, table, key, value):
+        """
+        Function to conditionally update a key-value pair in a lmdb table
+        that holds comma-separated strings as value. This function reads
+        the first of comma-seperated strings and then compares it with the
```
Comment (danintel) [inline shared_kv_storage/kv_storage/interface/kv_csv_interface.py:80]: s/seperated/separated/

File: shared_kv_storage/kv_storage/remote_lmdb/shared_kv_dbstore.py [hunk 2]
```diff
@@ -221,4 +221,38 @@ def csv_pop(self, table, key):
 
         return value
 
+# ---------------------------------------------------------------------------------------------------
+    def csv_match_pop(self, table, key, value):
+        """
+        Function to conditionally update a key-value pair in a lmdb table
+        that holds comma-separated strings as value. This function reads
+        the first of comma-seperated strings and then compares it with the
```
Comment (danintel) [inline shared_kv_storage/kv_storage/remote_lmdb/shared_kv_dbstore.py:229]: s/seperated/separated/

File: shared_kv_storage/db_store/db_store_csv.h [hunk 3]
```diff
@@ -82,4 +81,27 @@ class DbStoreCsv : public DbStore {
...
         // XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+        /**
+         * Updates a key->value pair in the database store conditionally.
+         * Reads the first string from value which is a comma separated
+         * string and removes it if the value passed in matches. The behavior
+         * is similar to peeking and then removing (if matched) from the head
+         * of a deque. Deletes the key-value pair if this is the lone element
+         * in the value.
+         *
+         * @param table_b64     base64 encode table name
```
Comment (danintel) [inline shared_kv_storage/db_store/db_store_csv.h:92]: s/encode/encoded/

File: shared_kv_storage/db_store/packages/lmdb_store.cpp [hunk 4]
```diff
@@ -497,29 +498,37 @@ tcf_err_t db_store::db_store_csv_pop(
...
+    if(isMatch && outValue != first_value){
+        return TCF_ERR_SYSTEM;
     }
 
     // If this is the only value, return value and delete key->value from db storage
     if(isLast){
         outValue = valueBuffer;
         return db_store_del(table, inId.data(), inId.size(), valueBuffer.data(), valueBuffer.size());
     }
+    outValue = first_value;
```
Comment (manojgop) [inline shared_kv_storage/db_store/packages/lmdb_store.cpp:531]: Please add a comment why valueBuffer is added back. -> ValueBuffer contains remaining csv values  (after deleting the popped value) which is added back

File: shared_kv_storage/kv_storage/remote_lmdb/shared_kv_dbstore.py [hunk 5]
```diff
@@ -221,4 +221,38 @@ def csv_pop(self, table, key):
...
+                            strings matches. None, otherwise.
+        """
+        try:
+            if key != "":
+                value = self._db_store.db_store_csv_match_pop(
+                    table, key, value)
+            else:
+                value = None
+        except Exception as ex:
+            logger.warn("Exception occurred in csv_match_pop - {}".
```
Comment (manju956) [inline shared_kv_storage/kv_storage/remote_lmdb/shared_kv_dbstore.py:250]: Can we change log level to error?

File: shared_kv_storage/db_store/packages/db_store_wrapper.h [hunk 6]
```diff
@@ -240,6 +241,7 @@ namespace db_store {
      * @param table      table name
      * @param inId       id byte array
      * @param outValue   data to be written
+     * @param isMatch    flag to indicate if a match is required before pop
```
Comment (manojgop) [inline shared_kv_storage/db_store/packages/db_store_wrapper.h:244]: You mean match of `id`

**Pull request merged**